program-runtime/src/loaded_programs.rs

```diff
@@ -356,6 +376,23 @@ impl LoadedPrograms {
         entry
     }
 
+    /// Insert all the entries from the iterator to the cache.
+    pub fn replenish_many(
+        &mut self,
+        programs: impl Iterator<Item = (Pubkey, Vec<Arc<LoadedProgram>>)>,
+    ) {
+        programs.for_each(|(key, updated)| {
+            let second_level = self.entries.entry(key).or_insert_with(Vec::new);
+            if second_level.is_empty() {
```

Comment:
Do we really gain much from this special case?

Reply:
I think it's a bit optimized. Otherwise we'll call replenish for every entry in the vector, which is more work.